### PR TITLE
SONARCS-558 Add a fake profile importer to workaround SONAR-6969

### PR DESCRIPTION
**Pull request by dbolkensteyn**

### CODE REVIEW

File: src/main/java/org/sonar/plugins/csharp/SonarLintFakeProfileImporter.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ */
+package org.sonar.plugins.csharp;
+
+import org.sonar.api.profiles.ProfileImporter;
+import org.sonar.api.profiles.RulesProfile;
+import org.sonar.api.utils.ValidationMessages;
+
+import java.io.Reader;
+
+// SONARCS-558 workaround for SONAR-6969
```
Comment (tvsonar) [inline src/main/java/org/sonar/plugins/csharp/SonarLintFakeProfileImporter.java]: @dbolkensteyn shouldn't we add a TODO or FIXME in this comment?

File: src/main/java/org/sonar/plugins/csharp/SonarLintFakeProfileImporter.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+  public SonarLintFakeProfileImporter() {
+    super("sonarlint-vs-cs-fake", "Do not use");
+    setSupportedLanguages(CSharpPlugin.LANGUAGE_KEY);
+  }
+
+  @Override
+  public RulesProfile importProfile(Reader reader, ValidationMessages messages) {
+    messages.addErrorText("Do not use this profile importer.");
+    return new RulesProfile(getName(), CSharpPlugin.LANGUAGE_KEY);
```
Comment (SonarTech) [inline src/main/java/org/sonar/plugins/csharp/SonarLintFakeProfileImporter.java]: MINOR Constructor 'RulesProfile(...)' is deprecated. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MINOR 1 minor

Watch the comments in this conversation to review them.

File: src/test/java/org/sonar/plugins/csharp/SonarLintProfileFakeImporterTest.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    assertThat(importer.getSupportedLanguages()).containsOnly("cs");
+    assertThat(importer.getName()).isEqualTo("Do not use");
+
+    RulesProfile profile = importer.importProfile(null, messages);
+    assertThat(messages.getErrors()).containsExactly("Do not use this profile importer.");
+    assertThat(profile.getName()).isEqualTo("Do not use");
+    assertThat(profile.getLanguage()).isEqualTo("cs");
+    assertThat(profile.getActiveRules()).isEmpty();
+  }
+
```
Comment (tvsonar) [inline src/test/java/org/sonar/plugins/csharp/SonarLintProfileFakeImporterTest.java]: @dbolkensteyn is it a best practice to leave an empty line here? I've seen this elsewhere too.

**Comment by tvsonar** [comment]
LGTM :-)

File: src/main/java/org/sonar/plugins/csharp/SonarLintFakeProfileImporter.java (hunk 1 above)
Comment (dbolkensteyn) [inline src/main/java/org/sonar/plugins/csharp/SonarLintFakeProfileImporter.java]: no, it's neither a todo nor a fixme - just a comment to say why this class exists

File: src/test/java/org/sonar/plugins/csharp/SonarLintProfileFakeImporterTest.java (hunk 3 above)
Comment (dbolkensteyn) [inline src/test/java/org/sonar/plugins/csharp/SonarLintProfileFakeImporterTest.java]: indeed

File: src/main/java/org/sonar/plugins/csharp/SonarLintFakeProfileImporter.java (hunk 1 above)
Comment (tvsonar) [inline src/main/java/org/sonar/plugins/csharp/SonarLintFakeProfileImporter.java]: Okay, but when SONAR-6969 is fixed, we'll likely want to drop this class.

Comment (dbolkensteyn) [inline src/main/java/org/sonar/plugins/csharp/SonarLintFakeProfileImporter.java]: SONAR-6969 is already fixed and ships in SQ 5.2.

**Pull request merged**